### PR TITLE
handle folder dropped onto tray icon in macOS

### LINKED ISSUE #5048: Spring Load Mac Dock Icon
**Is your feature request related to a problem? Please describe.**
I have project/repo/folder open in finder and I want to quickly open that repo in GitHub Desktop

**Describe the solution you'd like**
I want to be able to drag a folder to the GitHub Desktop dock icon which will open the corresponding repo in GitHub Desktop or offer to create it in GitHub Desktop if it doesn't already exist.

**Describe alternatives you've considered**
:|

**Teachability, Documentation, Adoption, Migration Strategy**
Many apps like VScode support this workflow.

### PR DESCRIPTION
**Pull request by shiftkey**
Fixes #5048 

Two parts to this PR:

 - registering with the OS that the app can handle dropping a folder _only_ - 🎩hat tip to https://github.com/zeit/hyper/pull/307 for figuring this bit out
 - handling the `open-file` event inside Electron and propagating that to the app

TODO:

 - [x] review `LSItemContentTypes` entries and only include extensions we know we should handle

### CODE REVIEW

**Comment by shiftkey** [comment]
Test installer for macOS is available [here](https://desktop.githubusercontent.com/releases/1.3.3-test8-4a0d8729/GitHubDesktop.zip) for anyone who would like to verify these changes.

File: script/info.plist [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	<array>
+		<dict>
+			<key>CFBundleTypeName</key>
+			<string>Folders</string>
+			<key>CFBundleTypeRole</key>
+			<string>Viewer</string>
+			<key>LSItemContentTypes</key>
+			<array>
+				<string>public.folder</string>
+				<string>com.apple.bundle</string>
```
[the remark this answers is not in the document]
Comment (j-f1) [reply script/info.plist]: Might want to keep the `resolvable` one (if it functions properly), but the other ones seem unnecessary. However, I have GitX installed, and it makes bare `foo.git` repos into a package, so we might need to add that back in if we start supporting bare repositories.

[the remark this answers is not in the document]
Comment (shiftkey) [reply script/info.plist]: @j-f1 thanks for the help!

>  and it makes bare `foo.git` repos into a package, so we might need to add that back in if we start supporting bare repositories.

For the moment we're tracking blocking bare repositories in #4293 but I'll add this context to that thread for future reference.

[the remark this answers is not in the document]
Comment (shiftkey) [reply script/info.plist]: For security reasons I ended up just sticking with `public.folder` support - I figure that's safer for the first ship, and if people find other things that aren't working as expected we can talk about and reproduce them down the track...

File: script/info.plist [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+			<key>LSItemContentTypes</key>
+			<array>
+				<string>public.folder</string>
+			</array>
+			<key>LSHandlerRank</key>
+			<string>Alternate</string>
+		</dict>
+	</array>
+</dict>
+</plist>
```
Comment (iAmWillShepherd) [inline script/info.plist]: Just need a final newline. I'll be adding it directly to this PR since you're out 👍

**Comment by tierninho** [comment]
### Areas tested

- [x] drag and drop single repo onto Desktop dock icon
- [x] drag and drop multiple repos onto Desktop dock icon
- [x] drag and drop onto Desktop dock icon while administrative modal present on Desktop
- [x] dragged repo out of trash onto Desktop dock icon
- [x] drag and drop onto Desktop dock icon while Desktop active/quit
- [x] drag and drop onto Desktop dock icon a new/existing repo
- [x] drag and drop onto Desktop desktop icon, not the dock icon
- [x] drag and drop an empty folder onto Desktop dock icon
- [x] drag and drop a non-repo onto Desktop dock icon 
- [x] use `esc` key to cancel drag and drag

## Notes
 - When dragging a repo from the trash and dropping it on an open application itself (current way), a modal asks you if you would like to add a Local Repo, thus giving you the opportunity to edit the path. However, if you drag and drop onto the dock icon, this modal does not surface. 

Overall the functionality works well despite the above edge case.

**Comment by shiftkey** [comment]
:apple:

**Comment by shiftkey** [comment]
@tierninho

> When dragging a repo from the trash and dropping it on an open application itself (current way), a modal asks you if you would like to add a Local Repo, thus giving you the opportunity to edit the path. However, if you drag and drop onto the dock icon, this modal does not surface.

I believe we treat these as separate flows. In #4513 we changed the default flow for `github [path]` to not prompt when the folder is a valid Git repository, and just select it in the app. That's the flow we use here too, because of the main process communicating to the `AppWindow`.

**Pull request merged**